File: 2025-2/sesion-de-clase-19/java/SoftProg/SoftProgNegocio/src/main/java/pe/edu/pucp/inf30/softprog/boimpl/rrhh/CuentaUsuarioBOImpl.java
package pe.edu.pucp.inf30.softprog.boimpl.rrhh;

import java.util.List;
import pe.edu.pucp.inf30.softprog.bo.rrhh.CuentaUsuarioBO;
import pe.edu.pucp.inf30.softprog.dao.rrhh.CuentaUsuarioDAO;
import pe.edu.pucp.inf30.softprog.daoimpl.rrhh.CuentaUsuarioDAOImpl;
import pe.edu.pucp.inf30.softprog.modelo.Estado;
import pe.edu.pucp.inf30.softprog.modelo.rrhh.CuentaUsuario;

/**
 *
 * @author eric
 */
public class CuentaUsuarioBOImpl implements CuentaUsuarioBO {
    private final CuentaUsuarioDAO cuentaUsuarioDao;
    
    public CuentaUsuarioBOImpl() {
        this.cuentaUsuarioDao = new CuentaUsuarioDAOImpl();
    }
    
    @Override
    public List<CuentaUsuario> listar() {
        return this.cuentaUsuarioDao.leerTodos();
    }

    @Override
    public CuentaUsuario obtener(int id) {
        return this.cuentaUsuarioDao.leer(id);
    }

    @Override
    public void eliminar(int id) {
        this.cuentaUsuarioDao.eliminar(id);
    }

    @Override
    public void guardar(CuentaUsuario modelo, Estado estado) {
        if (estado == Estado.Nuevo) {
            this.cuentaUsuarioDao.crear(modelo);
        }
        else {
            this.cuentaUsuarioDao.actualizar(modelo);
        }
    }

    @Override
    public boolean login(String userName, String password) {
        return this.cuentaUsuarioDao.login(userName, password);
    }
}

File: 2025-2/sesion-de-clase-19/dotnet/SoftProgNegocio/BOImpl/CuentaUsuarioBOImpl.cs
using System.Collections.Generic;
using PUCP.SoftProg.Modelo;
using PUCP.SoftProg.Modelo.RRHH;
using PUCP.SoftProg.Negocio.BO;
using PUCP.SoftProg.Persistencia.DAO.RRHH;
using PUCP.SoftProg.Persistencia.DAOImpl.RRHH;

namespace PUCP.SoftProg.Negocio.BOImpl {
    public class CuentaUsuarioBOImpl : ICuentaUsuarioBO {
        private readonly ICuentaUsuarioDAO cuentaUsuarioDAO;

        public CuentaUsuarioBOImpl() {
            this.cuentaUsuarioDAO = new CuentaUsuarioDAOImpl();
        }

        public void Eliminar(int id) {
            this.cuentaUsuarioDAO.Eliminar(id);
        }

        public void Guardar(CuentaUsuario cuentaUsuario, Estado estado) {
            if (estado == Estado.Nuevo) {
                this.cuentaUsuarioDAO.Crear(cuentaUsuario);
            }
            else if (estado == Estado.Modificado) {
                this.cuentaUsuarioDAO.Actualizar(cuentaUsuario);
            }
        }

        public List<CuentaUsuario> Listar() {
            return this.cuentaUsuarioDAO.LeerTodos();
        }

        public CuentaUsuario Obtener(int id) {
            return this.cuentaUsuarioDAO.Leer(id);
        }

        public bool Login(string userName, string password) {
            return this.cuentaUsuarioDAO.Login(userName, password);
        }
    }
}
